core/src/main/java/zipkin2/reporter/HttpEndpointSupplier.java

```diff
@@ -0,0 +1,84 @@
+package zipkin2.reporter;
+
+import java.util.List;
+
+/**
+ * HTTP-based {@link BytesMessageSender senders} use this to resolve a potentially-pseudo endpoint
+ * passed by configuration to a real endpoint.
+ *
+ * <p>Senders should consider the special value {@link #FIXED_FACTORY} and the type {@link Fixed} to
+ * avoid dynamic lookups when constants will be returned.
+ *
+ * <p>Senders are not called during production requests, rather in time or size bounded loop, in a
+ * separate async reporting thread. Implementations that resolve endpoints via remote calls, such as
+ * from Eureka, should cache internally to avoid blocking the reporter thread on each loop.
+ *
+ * <p>Some senders, such as Armeria, may have more efficient and precise endpoint group logic. In
```

Comment:
For this part, I might put an example in the test tree. While this will pull armeria into our deps that are analyzed with snyk etc, usually there are no CVEs that aren't resolved quickly. Also the dependency tree isn't as huge as spring boot, so it isn't as bad for laptops in hotels.